doc/releases/changelog-dev.md

```diff
@@ -407,6 +407,52 @@
 
 <h3>Improvements</h3>
 
+* `qml.dot` now groups coefficients together.
+  [(#3691)](https://github.com/PennyLaneAI/pennylane/pull/3691)
+
+  ```pycon
+  >>> qml.dot(coeffs=[2, 2, 2], ops=[qml.PauliX(0), qml.PauliY(1), qml.PauliZ(2)])
+  2*(PauliX(wires=[0]) + PauliY(wires=[1]) + PauliZ(wires=[2]))
```

Comment:
Is this a merge conflict?

Reply:
Nope, these features were needed for this PR to work. I didn't create a PR for each of those because they were only 4-5 lines each.